src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```
@@ -1321,9 +1343,78 @@ private boolean performDecoderControl(DecoderInfo decoderInfo, String controlURL
+		try {
+			if (this.authenticationCookie.getSessionID() != null) {
+				String request = decoderInfo.jsonRequest();
+				if (logger.isDebugEnabled()) {
+					logger.debug(request);
+				}
+				DecoderData decoderData = doPut(buildDeviceFullPath(DecoderURL.BASE_URI + DecoderURL.DECODERS + DecoderConstant.SLASH + decoderID + controlURL), request, DecoderData.class);
+				if (decoderData == null) {
+					populateActiveControlAfterControlFailed(decoderInfo);
+					addAdvanceControlProperties(advancedControllableProperties, createSwitch(stats, decoderControllingGroup + DecoderControllingMetric.STATE.getName(), decoderInfo.getState().isRunning(),
+							DecoderConstant.OFF, DecoderConstant.ON));
+					throw new ResourceNotReachableException(DecoderConstant.DECODER_CONTROL_ERR + controlMethod);
+				}else{
+					this.decoderInfoDTOList.set(Integer.parseInt(decoderID), decoderData.getDecoderInfo());
+					this.localDecoderInfoList.set(Integer.parseInt(decoderID), decoderData.getDecoderInfo());
+					populateDecoderControl(stats, advancedControllableProperties, Integer.parseInt(decoderID));
+				}
+			} else {
+				populateActiveControlAfterControlFailed(decoderInfo);
+				addAdvanceControlProperties(advancedControllableProperties, createSwitch(stats, decoderControllingGroup + DecoderControllingMetric.STATE.getName(), decoderInfo.getState().isRunning(),
+						DecoderConstant.OFF, DecoderConstant.ON));
+				throw new ResourceNotReachableException(DecoderConstant.DECODER_CONTROL_ERR + controlMethod);
+			}
+		} catch (Exception e) {
+			populateActiveControlAfterControlFailed(decoderInfo);
+			addAdvanceControlProperties(advancedControllableProperties, createSwitch(stats, decoderControllingGroup + DecoderControllingMetric.STATE.getName(), decoderInfo.getState().isRunning(),
+					DecoderConstant.OFF, DecoderConstant.ON));
+			throw new ResourceNotReachableException(DecoderConstant.DECODER_CONTROL_ERR + controlMethod + DecoderConstant.NEXT_LINE + e.getMessage());
```

the same issue like above, please do not swallow the exception, add the "e" like second arg

---

i fixed it